pkg/devicerepository/grpc.go

```diff
@@ -60,13 +61,31 @@ func (dr *DeviceRepository) ensureBaseAssetURLs(models []*ttnpb.EndDeviceModel)
 
 const defaultLimit = 1000
 
-func requireApplication(ctx context.Context, ids *ttnpb.ApplicationIdentifiers) error {
-	return rights.RequireApplication(ctx, ids, ttnpb.Right_RIGHT_APPLICATION_DEVICES_READ)
+var errNotAuthenticated = errors.DefineUnauthenticated(
+	"no_rights_in_context",
+	"Not authenticated, need to be logged in to make this request",
```

Comment:
The error description should not start with upper case letter, and can be shorter: `unauthenticated`.